content/new-routing.md

```diff
@@ -0,0 +1,446 @@
+
+```js
+FlowRouter.setParams({_id: newList._id});
+```
+
+Of course, it is more general to call `FlowRouter.go()`, so unless you are being very specific in what you are doing it's usually better to use that.
+
+### Storing data in the URL
+
+As we discussed in the introduction, the URL is really just a serialization of some part of the client-side state the user is looking at. Although parameters can only be strings, it's possible to convert any type of data to a string via serializing it.
+
+In general if you want to store arbitrary serializable data in a URL param, you can use `EJSON.stringify()` to turn it onto a string. You'll need to URL-encode the string as well to remove any characters that have meaning in a URL:
+
+```js
+FlowRouter.setQueryParams({data: encodeURIComponent(EJSON.stringify(data))});
+```
+
+You can then get the data back out of Flow Router in the opposite way (note that Flow Router unescapes the dat for you automatically):
+
+```js
+const data = EJSON.parse(FlowRouter.getQueryParam('data'));
+```
+
+## Redirecting
+
+Sometimes, your users will end up on a page that isn't the best place for them to be. Maybe the data they were looking for has moved, maybe they were on an admin panel page and logged out, or maybe they just created a new object and you want them to end up on the page for the thing they just created.
+
+Usually, we can redirect in response to a user's action by calling `FlowRouter.go()` and friends, like in our list creation example above, but if a user browses directly to a URL that doesn't exist, it's useful to know how to redirect immediately.
+
+If a URL is simply out-of-date (sometimes you can the URL scheme of an application), you can redirect inside the `action` function of the route:
+
+```js
+FlowRouter.route('/old-list-route/:_id', {
+  action(params) {
+    FlowRouter.go('listsShow', params);
+  }
+});
+```
+
+### Redirecting dynamically
+
+If however, you need some data to redirect, you'll need to render part of the component hierarchy, as that is the place where data subscribing happens. For example, in the Todos example app, we want to make the root (`/`) route redirect to the first known list. To achieve this, we need to render a special `rootRedirector` route:
+
+```js
+FlowRouter.route('/', {
+  name: 'home',
+  action: () => {
+    BlazeLayout.render('appBody', {main: 'rootRedirector'});
+  }
+});
+```
+
+Because the `rootRedirector` template is rendered inside the `appBody` layout which takes care of subscribing to the set of lists the user knows about *before* rendering it's sub-template, and we are guaranteed there is at least one such list, we can simply do:
+
+```js
+Template.rootRedirector.onCreated(() => {
+  // We need to set a timeout here so that we don't redirect from inside a redirection
```

Comment:
It's actually there in the last commit: https://github.com/meteor/guide/commit/63055c313562e29fd8e97cb0964db9d4f602ed0f#diff-78e5e59bc4e54f9fd23edeca8488a375R336.